core/src/main/kotlin/ch/ergon/dope/resolvable/expression/CaseExpression.kt

```diff
@@ -0,0 +1,93 @@
+package ch.ergon.dope.resolvable.expression
+
+import ch.ergon.dope.DopeQuery
+import ch.ergon.dope.resolvable.WhenThenCondition
+import ch.ergon.dope.validtype.ValidType
+
+sealed class CaseExpression(
+    private val unaliasedExpression: UnaliasedExpression<out ValidType>? = null,
+    private val whenThenCondition: WhenThenCondition,
+    private vararg val additionalWhenThenConditions: WhenThenCondition,
+    private val elseCase: UnaliasedExpression<out ValidType>? = null,
+) : Expression {
+    internal fun getCaseExpressionDopeQuery(alias: String? = null): DopeQuery {
+        val unaliasedExpressionDopeQuery = unaliasedExpression?.toDopeQuery()
+        val whenThenConditionDopeQuery = whenThenCondition.toDopeQuery()
+        val additionalWhenThenConditionsDopeQuery = additionalWhenThenConditions.map { it.toDopeQuery() }
+        val elseCaseDopeQuery = elseCase?.toDopeQuery()
+
+        val queryString = buildString {
```

Comment:
to be honest, I think it's less readable with this string-builder, would simple string cocatenation:

```
        val queryString =
            """
                CASE ${unaliasedExpressionDopeQuery?.queryString ?: ""} 
                ${whenThenConditionDopeQuery.queryString}
                ...                
            """.trimIndent()
```